lms/djangoapps/instructor_task/tests/test_tasks_helper.py

```diff
@@ -1965,10 +1964,9 @@ def test_grade_report(self):
                         u'Username': self.student.username,
                         u'Grade': '0.13',
                         u'Homework 1: Subsection': '0.5',
-                        u'Homework 2: Hidden': u'Not Attempted',
-                        u'Homework 3: Unattempted': u'Not Attempted',
-                        u'Homework 4: Empty': u'Not Attempted',
-                        u'Homework (Avg)': '0.125',
+                        u'Homework 3: Unattempted': None,
```

Comment:
Quick question here on why the values of the remaining test cases needed to be changed?

Reply:
@fysheets I am not really sure why this needs to be changed in the tests. I created a similar scenario locally and generated a grade report for it and it showed "Not attempted",  but in the test it shows None so I am thinking the test has been setup in such a way. This is also the reason why I wanted to create a sandbox and generate a report grade. Just to double check everything is working as expected.

Reply:
It's a combination of two things:
1) This line https://github.com/edx/edx-platform/blob/master/lms/djangoapps/instructor_task/tasks_helper/grades.py#L121 makes it so that we number subsections successively in the report header.  Since we're now skipping over hidden, graded subsections, the "Unattempted" subsection moves from 3 -> 2, and the "Empty" subsection moves from 4 -> 3.
2) The `ignore_other_columns` kwargs makes `verify_rows_in_csv()` ignore any column headers that aren't in the actual report.
I fixed this up in the latest force-pushed commit.

Reply:
And for the sake of completeness, the HW average grade changed from 1/8 to 1/6, because we're no longer incorporating the hidden subsection into the average score of homework.